census_api_function/census_api_function.R

```diff
@@ -0,0 +1,161 @@
+# function for pulling clean acs data: clean_acs
+# Author: David Whyman
+# Date: Wed Jul 03 11:30:00 2019
+# SET UP ==============================================
+
+pkgs <- c("tidyverse", "tidycensus","skimr")
+
+check <-
+  sapply(pkgs,
+         require,
+         warn.conflicts = TRUE,
+         character.only = TRUE)
+
+if (any(!check)) {
+  pkgs.missing <- pkgs[!check]
+  install.packages(pkgs.missing)
+  check <-
+    sapply(pkgs.missing,
+           require,
+           warn.conflicts = TRUE,
+           character.only = TRUE)
+}
+
+census_api_key("e5ae735f65e6b93f5c2f9bc16bea67b72fa202da")
+
+# FUNCTION: CLEAN_ACS ==============================================
+
+# unlike tidycensus's get_acs, clean_acs assigns meaningful variable names
+
+# first argument = geography
+# second argument = variable codes
+# third argument = endyear of survey (yyyy)
+# fourth argument = number of years survey spans (either 1, 3, or 5)
+
+
+
+clean_acs <- function(geography, variables, year, span) {
+  
+  geogr <- geography
+  vari <- variables
+  ye <- year
+  spann <- span
+  
+  geogr_name <- paste0(enquo(geogr),"_name")
+  varkey <- c("0")
+  x <- c("0")
+  
+  # load libraries
+  library(tidycensus)  
+  library(tidyverse)  
+  
+  
+  # pull data with get acs
+  x <- get_acs(
+    geography = geogr,
+    variables = vari,
+    cache_table = TRUE,
+    year = ye,
+    survey = paste0("acs", enquo(spann))[2]
+  )
+  
+  # load descriptive labels for subject tables OR detailed tables
+  suppressWarnings(
+    if (str_detect(variables, "S")) {
+      varkey <-
+        load_variables(year, paste0("acs", enquo(span), "/subject")[2], cache = TRUE)
+    } else {
+      varkey <-
+        load_variables(year, paste0("acs", enquo(span))[2], cache = TRUE)
+    }
+  )
+  
+  # clean pulled data and rename columns to descriptive labels 
+  final <- varkey %>%
+    filter(name %in% x$variable) %>%
+    left_join(x, by = c("name" = "variable")) %>%
+    select(-concept) %>%
+    gather(key = "measure", value = "values", estimate, moe) %>%
+    mutate(label = tolower(str_replace_all(label, "[[:punct:]]", " "))) %>%
+    mutate(label = str_remove(label, "estimate")) %>%
+    mutate(label = str_replace_all(label, " ", "_")) %>%
+    unite(label, label, measure, sep = "_") %>%
+    unite(label, name, label, sep = "_") %>%
+    spread(key = label, value = values) %>%
+    rename("geoid" = GEOID,
+           geography_name = "NAME") %>%
+    select_if(~sum(!is.na(.)) > 0)
+  
+  
+  return(final)
+  
+}
+
+# VARIABLE CODES =================================================
+
+#variable codes for poverty rate by race (manual lookup & test)
+pov_race_codes<-c("S1701_C03_013",
```

Comment:
you can use ` purrr::map_chr(seq(13,21),function(x)paste0("S1701_C03_0",x)) ` to 1) simplify script 2) minimize copy-pasting mistakes